### ISSUE TITLE
[BUG] Not sure the plot_train_test_folds 0-train 1-test encoding is correct in CombinatorialPurgedCV

### ISSUE THREAD

**Issue by CarloNicolini** [skfolio/skfolio#32]
**Describe the bug**
The title of the CombinatorialPurgedCV.plot_train_test_folds() is probably misleading.
Otherwise it happens that the portfolio is fitted with very small training sets and tested on very long ones.

**To Reproduce**

```python
from skfolio.model_selection import CombinatorialPurgedCV
CombinatorialPurgedCV().plot_train_test_folds()
```

**Expected behavior**
I believe train is 1 and test is 0.

**Additional context**
![image](https://github.com/skfolio/skfolio/assets/1758847/544c9853-76e6-41ab-9042-2147a49fa1f7)

**Versions**
0.1.1

I also did an analysis based on this encoding and the results are similar:
![image](https://github.com/skfolio/skfolio/assets/1758847/69872556-542e-4043-b9ab-a5a60f112937)

**Comment by HugoDelatte** [comment]
Thanks Carlo, I had a look and  `CombinatorialPurgedCV` uses ` n_folds=10` and `n_test_folds=8` as default parameters. That implies `n_train_folds=2` which is what the graph is showing. But I agree that's not expected, but the problem comes from the default: maybe we should use `n_test_folds=2` and ` n_folds=5` instead (same as the initial paper). Also, the fact that the graph orders the combinations to start with test in fold 0,1.. and not train is not intuitive.

**Comment by CarloNicolini** [comment]
Otherwise add a parameter `n_train_folds`, together with `n_test_folds` and `n_folds` with some checking of correctness when all three of them are passed. 
So for example what would you suggest to set for a dataset of 20 years of daily data if I want train set to be 5 times the test set?
Judging from the output of `.summary` now I was training on 1008 days

```python
import numpy as np
from skfolio.model_selection import CombinatorialPurgedCV
CombinatorialPurgedCV().summary(np.arange(20*252))
```
````
Number of Observations             5040
Total Number of Folds                10
Number of Test Folds                  8
Purge Size                            0
Embargo Size                          0
Average Training Size              1008
Number of Test Paths                 36
Number of Training Combinations      45
dtype: int64
````

**Comment by HugoDelatte** [comment]
I think that because only two params are necessary, the API should probably also have 2 params to reduce confusion. So either:
* (`n_folds`, `n_test_folds`)
* (`n_folds`, `n_train_folds`)
* (`n_train_folds`, `n_test_folds`)

The choice of (`n_folds`, `n_test_folds`) is consistent with the initial paper that uses 5 total folds and 2 tests folds (so implying 3 train folds).

To answer your second question, you can find an infinity of combinations that will give you roughly tain_size=5*test_size, the second parameter to control is the number of paths. 

Usually, when you cross validate your portfolio model, you want to control for two things: the training size and the number of paths generated. So I think that an helper function giving you the optimal (`n_folds`, `n_test_folds`) based on `training_size` and `n_paths` will be useful. I already have smth working, I can create a PR to expose it.

**Comment by CarloNicolini** [comment]
Exactly, it's an underspecified problem at the end. I also thought about an helper function to help the user devise the correct parameters to be passed to `CombinatorialCV`.

**Comment by HugoDelatte** [comment]
Work in progress in [this branch](https://github.com/skfolio/skfolio/blob/08a7a7df5addc48d704d6806f503e8337aa9bc79/src/skfolio/model_selection/_combinatorial.py#L506)
